contracts/account/extensions/AccountERC6900.sol

```diff
@@ -0,0 +1,494 @@
+            ManifestExecutionHook memory executionHook = executionHooks[i];
+            bytes4 executionSelector = executionHook.executionSelector;
+            // module
+            uint32 entityId = executionHook.entityId;
+            bool isPreHook = executionHook.isPreHook;
+            bool isPostHook = executionHook.isPostHook;
+            Execution storage execution = _executions[executionSelector];
+            require(
+                execution.executionHooks.add(bytes32(abi.encodePacked(module, entityId, isPreHook, isPostHook))),
+                ERC6900Utils.ERC6900AlreadySetExecutionHookForExecution()
+            );
+        }
+        // The account SHOULD add all supported interfaces as specified in the manifest.
+        bytes4[] memory interfaceIds = manifest.interfaceIds;
+        for (uint256 i = 0; i < interfaceIds.length; i++) {
+            bytes4 interfaceId = interfaceIds[i];
+            require(!_interfaceIds[interfaceId], "Interface already set"); // TODO use custom error
+            _interfaceIds[interfaceId] = true;
+        }
+        // The account SHOULD call onInstall on the execution module to initialize state if specified by the user.
+        IERC6900Module(module).onInstall(installData);
+        // The account MUST emit ExecutionInstalled as defined in the interface for all installed executions.
+        emit ExecutionInstalled(module, manifest);
+    }
+
+    /**
+     * @dev Uninstalls a module.
+     *
+     *
+     * Requirements:
+     *
+     * * TODO
+     */
+    function _uninstallModule(uint256 moduleTypeId, address module, bytes memory deInitData) internal virtual {
+        // TODO
+        // IERC6900Module(module).onUninstall(deInitData);
+        // emit ModuleUninstalled(moduleTypeId, module);
+    }
+
+    /**
+     * @dev Fallback function that delegates the call to the installed handler for the given selector.
+     *
+     */
+    function _fallback() internal virtual returns (bytes memory) {
+        // TODO
+    }
+
+    /// @dev Returns the fallback handler for the given selector. Returns `address(0)` if not installed.
+    function _fallbackHandler(bytes4 selector) internal view virtual returns (address) {
+        return _fallbacks[selector];
+    }
+
+    function _decodeValidationConfig(
+        ValidationConfig validationConfig
+    ) internal pure virtual returns (address module, uint32 entityId, bytes1 validationFlags) {
+        return (
+            address(Packing.extract_32_20(ValidationConfig.unwrap(validationConfig), 0)),
+            uint32(Packing.extract_32_4(ValidationConfig.unwrap(validationConfig), 20)),
+            Packing.extract_32_1(ValidationConfig.unwrap(validationConfig), 21)
+        );
+    }
+
+    /**
+     * @dev Extracts the validator from the user operation.
+     *
+     */
+    function _extractUserOpValidator(PackedUserOperation calldata userOp) internal pure virtual returns (ModuleEntity) {
```

Comment:
`_extractSignatureValidator` already does that. Maybe we can drop this function, and only use `_extractSignatureValidator`